scripts/scil_screenshot_volume_mosaic_overlap.py

```diff
@@ -0,0 +1,240 @@
+  brain_mask.nii.gz \
+  mosaic_overlap_tissue_map.png \
+  30 40 50 60 70 80 90 100 \
+  2 4 \
+  --vol_cmap_name plasma
+
+python scil_screenshot_volume_mosaic_overlap.py \
+  t1.nii.gz \
+  brain_mask.nii.gz \
+  mosaic_overlap_t1_tisue_map.png \
+  30 40 50 60 70 80 90 100 \
+  2 4 \
+  --in_labelmap tissue_map.nii.gz \
+  --axis_name sagittal \
+  --labelmap_cmap_name viridis
+"""
+
+import argparse
+
+import nibabel as nib
+from dipy.io.utils import is_header_compatible
+
+from scilpy.io.utils import (
+    axis_name_choices,
+    add_overwrite_arg,
+    assert_inputs_exist,
+    assert_outputs_exist,
+    ranged_type,
+)
+from scilpy.image.utils import check_slice_indices
+from scilpy.viz.scene_utils import (
+    check_mosaic_layout, compose_mosaic, screenshot_slice,
+)
+
+
+def _build_arg_parser():
+
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter
+    )
+
+    # Positional arguments
+    p.add_argument("in_vol", help="Input volume image file.")
+    p.add_argument("in_transparency_mask", help="Input mask image file.")
+    p.add_argument(
+        "out_fname",
+        help="Name of the output image mosaic (e.g. mosaic.jpg, mosaic.png)."
+    )
+    p.add_argument(
+        "slice_ids", nargs="+", type=int, help="Slice indices for the mosaic."
+    )
+    p.add_argument(
+        "mosaic_rows_cols",
+        nargs=2,
+        # metavar=("rows", "cols"),  # CPython issue 58282
+        type=int,
+        help="The mosaic row and column count."
+    )
+
+    # Optional arguments
+    p.add_argument("--in_labelmap",  help="Labelmap image.")
+    p.add_argument(
+        "--axis_name",
+        default="axial",
+        type=str,
+        choices=axis_name_choices,
+        help="Name of the axis to visualize. [%(default)s]"
+    )
+    p.add_argument(
+        "--overlap_factor",
+        nargs=2,
+        metavar=("OVERLAP_HORIZ", "OVERLAP_VERT"),
+        default=(0.6, 0.0),
+        type=ranged_type(float, 0.0, 1.0),
+        help="The overlap factor with respect to the dimension. [%(default)s]"
+    )
+    p.add_argument(
+        "--win_dims",
+        nargs=2,
+        metavar=("WIDTH", "HEIGHT"),
+        default=(768, 768),
+        type=int,
+        help="The dimensions for the vtk window. [%(default)s]"
+    )
+    p.add_argument(
+        "--vol_cmap_name",
+        default=None,
+        help="Colormap name for the volume image data. [%(default)s]"
+    )
+    p.add_argument(
+        "--labelmap_cmap_name",
+        default="viridis",
+        help="Colormap name for the labelmap image data. [%(default)s]"
+    )
+
+    add_overwrite_arg(p)
+
+    return p
+
+
+def _parse_args(parser):
+
+    args = parser.parse_args()
+
+    inputs = []
+    output = []
+
+    inputs.append(args.in_vol)
+    inputs.append(args.in_transparency_mask)
+
+    if args.in_labelmap:
+        inputs.append(args.in_labelmap)
+
+    output.append(args.out_fname)
```

Comment:
I'm not against having this method (of course since I wrote both examples Jon cites). Can be particularly useful when the number of optional arguments grows bigger and there starts being dependencies between some of them. Same goes for `_get_data_from_inputs`. If you look in [`sci_visualize_fodf.py`](https://github.com/scilus/scilpy/blob/933ae303ca51ba2f95e1da3cf6a93a5a4f6f6c83/scripts/scil_visualize_fodf.py#L169) it turned into a pretty big method that does simplify the `main` in my opinion.

But it is true that for the actual script it does not simplify the `main` that much (at least for now, I don't know if there are plans to add other options in the future).